theseus_gui/src/pages/Settings.vue

```diff
@@ -1,7 +1,471 @@
+          Post Launch
+          <input ref="javaArgs" v-model="settings.hooks.post_exit" type="text" class="input" />
+        </div>
+      </div>
+    </Card>
+    <Button color="primary" class="save-btn" @click="saveSettings"><SaveIcon />Save changes</Button>
   </div>
 </template>
+
+<style lang="scss">
+.active {
+  background: var(--color-brand-highlight) !important;
+}
+
+.concurrent-downloads {
+  min-height: 40px;
+  max-width: 4.2rem !important;
+  text-align: center;
+}
+
+.auto-detect-modal {
+  .modal-body {
+    width: 45rem !important;
+    .content {
+      padding: 1rem;
+
+      button {
+        width: 100%;
+        display: flex;
+        align-items: center;
+        justify-content: center;
+
+        span {
+          display: inherit;
+          align-items: center;
+          justify-content: center;
+        }
+      }
+    }
+  }
+}
+
+.save-btn {
+  margin: 1rem;
+}
+
+.slider-input {
+  width: 5rem !important;
+  flex-basis: 5rem !important;
+}
+
+.installation-input {
+  width: 100% !important;
+  flex-grow: 1;
+}
+
+.theming,
+.settings-card {
+  margin: 1rem;
+  display: flex;
+  flex-direction: column;
+}
+
+.theme-dropdown {
+  text-transform: capitalize;
+}
+
+.settings-card {
+  display: flex;
+  flex-direction: column;
+  gap: 1rem;
+}
+
+.settings-title {
+  color: var(--color-contrast);
+}
+
+.settings-group {
+  display: flex;
+  flex-direction: column;
+  gap: 0.5rem;
+}
+
+.installation-input {
+  width: 100%;
+}
+
+.installation-buttons {
+  display: flex;
+  flex-direction: row;
+  flex-wrap: wrap;
+  align-items: center;
+  gap: 0.5rem;
+  margin: 0;
+}
+
+.sliders {
+  display: flex;
+  flex-wrap: wrap;
+  flex-direction: row;
+  gap: 1rem;
+  width: 100%;
+
+  .slider {
+    flex-grow: 1;
+  }
+}
+
+.toggle-setting {
+  display: flex;
+  flex-direction: row;
+  justify-content: space-between;
+  align-items: center;
+  gap: 0.5rem;
+}
+
+.card-divider {
+  background-color: var(--color-button-bg);
+  border: none;
+  color: var(--color-button-bg);
+  height: 1px;
+  margin: var(--gap-sm) 0;
+}
+
+.table-container {
+  display: grid;
+  grid-template-rows: repeat(auto-fill, auto);
+  width: 100%;
+  border-radius: var(--radius-md);
+  overflow: hidden;
+}
+
+.table-row {
+  display: grid;
+  grid-template-columns: 1fr 4fr 1.5fr;
+}
+
+.table-head {
+  .table-cell {
+    background-color: var(--color-accent-contrast);
+  }
+}
+
+.table-cell {
+  padding: 1rem;
+  height: 100%;
+  align-items: center;
+  vertical-align: center;
+  display: flex;
+}
+
+.table-text {
+  overflow: hidden;
+  white-space: nowrap;
+  text-overflow: ellipsis;
+  display: flex;
+
+  span {
+    display: inline-block;
+    text-overflow: ellipsis;
+    overflow: hidden;
+  }
+}
+
+.manage {
+  display: flex;
+  gap: 0.5rem;
+}
+
+.table-row:nth-child(even) .table-cell {
+  background-color: var(--color-bg);
+}
+
+.loading {
+  display: flex;
+  align-items: center;
+  justify-content: center;
+  height: 100% !important;
+}
```

Comment:
this page doesn't need a seperate loading animation- we will handle async pages eventually with a loading bar component similar to knossos

Reply:
Without the loading animation, the page tries to throw errors with the java 8 and java 17 settings being undefined (which is weird considering the onBeforeMount call). I'd like to leave it for now until we either change what Tauri returns as the default(empty) jres or we get the async pages in.